wolfSSL/src/chapter06.md

```diff
@@ -185,3 +185,158 @@ These contexts may be pointers to any user-specified context, which will then in
+
+To register a cryptographic callback function with wolfCrypt, use the `wc_CryptoCb_RegisterDevice` API. This takes a (`devId`), the callback function, and an optional user context that will be passed through to the callback when it is invoked.
+
+```
+typedef int (*CryptoDevCallbackFunc)(int devId, wc_CryptoInfo* info, void* ctx);
+
+WOLFSSL_API int wc_CryptoCb_RegisterDevice(int devId,
+                                           CryptoDevCallbackFunc cb,
+                                           void* ctx);
+```
+
+
+#### 3. Pass the `devId` as an argument to a wolfCrypt API function
+
+For wolfCrypt API’s use the init functions that accept `devId` such as:
+```
+wc_InitRsaKey_ex
+wc_ecc_init_ex
+wc_AesInit
+wc_InitSha256_ex
+wc_InitSha_ex
+wc_HmacInit
+wc_InitCmac_ex
+```
+This is not an exhaustive list. Please refer to the wolfCrypt API documentation to see if a particular algorithm supports crypto callbacks. 
+
+#### 4. (TLS only): associate the devId with a wolfSSL context
+To enable use of a crypto callback when using TLS, you must supply the `devId` arguments on initialization of a `WOLFSSL_CTX` or `WOLFSSL` struct.
+```
+wolfSSL_CTX_SetDevId(ctx, devId);
+wolfSSL_SetDevId(ssl, devId);
+```
+All relevant crypto for TLS connections associated with those structures will now invoke the crypto callback.
+
+### Writing your crypto callback
+
+When a crypto callback is invoked, it will need to know what cryptographic operation has been requested, as well as the location of the input and output data and how to communicate success/failure back to wolfCrypt. This information is relayed to the callback through the `wc_CryptoInfo* info` argument.
+
+At a high level, a crypto callback should determine if it supports the requested operation, act on the input data, update the `wc_CryptoInfo *info` argument with any relevant output data, and return a valid wolfCrypt error code.
+
+#### Determining the Type of Request
+
+The `wc_CryptoInfo` structure contains a union of different structures (some of which are unions themselves), each corresponding to a specific type of cryptographic operation. The type of operation to be performed is determined by `info->algo_type`, which should be a variant of `enum wc_AlgoType` defined in `wolfssl/wolfcrypt/types.h`.  Based on this `algo_type`, the appropriate union element within `wc_CryptoInfo` can be accessed to get or set parameters specific to the operation. We can refer to this top-level union as the "algo type union".
```

Comment:
Add comment about `HAVE_ANONYMOUS_INLINE_AGGREGATES` if getting error about compiler not allowing anonymous unions.